src/plots/cartesian/axes.js

```diff
@@ -611,31 +613,50 @@ axes.calcTicks = function calcTicks(ax) {
     generateTicks();
 
     if(ax.rangebreaks) {
-        var nTicksBefore = tickVals.length;
+        // replace ticks inside breaks that would get a tick
+        if(ax.tickmode === 'auto') {
+            for(var t = 0; t < tickVals.length; t++) {
+                var value = tickVals[t].value;
+                if(ax.maskBreaks(value) === BADNUM) {
+                    // find which break we are in
+                    for(var k = 0; k < ax._rangebreaks.length; k++) {
+                        var brk = ax._rangebreaks[k];
+                        if(value >= brk.min && value < brk.max) {
+                            // replace with break end
+                            tickVals[t] = {
+                                minor: false,
+                                value: brk.max
```

Comment:
I suspect just replacing the value with `brk.max` is going to get us in trouble at some point, ie when the end of the break would get a label that's different from the position within the break where we actually wanted the tick to be. Often the tick label formatting is sufficiently rounded that it will be correct, but this need not always be the case.

Can we
- keep track of the actual value that was requested in addition - ie do something like `tickVals[t] = {minor: false, value: brk.max, requestedValue: value}`
- when we reduce to one tick within the break, keep only the one with the largest requested value
- make sure that even if it's `brk.max` that's used to *position* the tick, we use `requestedValue` to generate its label.

Reply:
@alexcjohnson 
Good call. Done in https://github.com/plotly/plotly.js/pull/4677/commits/7a1c649bfc71b55f4ca22be37ec44067bd19c083, https://github.com/plotly/plotly.js/pull/4677/commits/d1ba367c00743c740db4116ab29c0ba62601ffff and https://github.com/plotly/plotly.js/pull/4677/commits/e718e3e05d382b0736014638ca31bdc8aa9eaf23.